core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -264,10 +264,10 @@ private[spark] class ExecutorAllocationManager(
     updateAndSyncNumExecutorsTarget(now)
 
     removeTimes.retain { case (executorId, expireTime) =>
-      val expired = now >= expireTime
+      var expired = now >= expireTime
       if (expired) {
-        initializing = false
-        removeExecutor(executorId)
+        expired = removeExecutor(executorId)
+        if (expired)  initializing = false
```

Comment:
I have done the style check before committing. sorry for missing the ```if``` block here.  I will fix that.